wgpu-native/src/command/render.rs

```diff
@@ -406,6 +407,38 @@ pub extern "C" fn wgpu_render_pass_set_stencil_reference(pass_id: RenderPassId,
     }
 }
 
+#[no_mangle]
+pub extern "C" fn wgpu_render_pass_set_viewport(
+    pass_id: RenderPassId,
+    x: f32,
+    y: f32,
+    w: f32,
+    h: f32,
+    min_depth: f32,
+    max_depth: f32,
+) {
+    let mut pass_guard = HUB.render_passes.write();
+    let pass = &mut pass_guard[pass_id];
+
+    unsafe {
+        use std::convert::TryFrom;
+        use std::i16;
+
+        pass.raw.set_viewports(
+            0,
+            &[hal::pso::Viewport {
+                rect: hal::pso::Rect {
+                    x: i16::try_from(x.round() as i64).unwrap_or(0),
```

Comment:
I guess WebGPU needs to spec rounding behavior here. Interestingly WebGL uses integers for x/y/width/height https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/viewport

Reply:
Agreed, although I find the specification to be trivial in this case:
  - viewport can be floating point because it's just a multiplication of the vertex-output clip space coordinate
  - after viewport transform, it's up to the pixel center to be either in or out the window, which is semantically equivalent to rounding here